src/pint/models/timing_model.py

```diff
@@ -545,9 +545,30 @@ def __setattr__(self, name, value):
         """Mostly this just sets ``self.name = value``.  But in the case where they are both :class:`Parameter` instances
         with different names, this copies the ``quantity``, ``uncertainty``, ``frozen`` attributes only.
         """
+        # print(name, value)
+        # try:
+        #     print(getattr(self, name).__class__)
+        # except:
+        #     pass
+        # try:
+        #     print(f"{name} in params: {name in self.params}")
+        # except:
+        #     print(f"Cannot tell if {name} is in params")
+        # if hasattr(self, "top_level_params"):
+        #     print(f"{self.top_level_params}")
         if isinstance(value, (Parameter, prefixParameter)) and name != value.name:
             for p in ["quantity", "uncertainty", "frozen"]:
                 setattr(getattr(self, name), p, getattr(value, p))
+        elif isinstance(value, u.Quantity):
+            log.warning(
+                f"Setting '{name}.quantity' to '{value}' although 'quantity' not specified"
+            )
+            getattr(self, name).quantity = value
+        elif isinstance(value, (float, str, bool, int)) and name != "name":
```

Comment:
This can also be a Time object.

Reply:
I think a bit more validation is needed here. Right now this will allow one to set a string as the value of a floatParameter, if I understand correctly. I think that is a potential foot gun.

Reply:
Time objects added

Reply:
I think your validation comment is handled by the `@setter` methods already.  e.g., `m.F0.value = 'abc'` is prevented already, and this works here too

Reply:
similarly, a quantity with the wrong units will be prevented.

Reply:
Ok...